src/main/java/solvd/ermakovich/ct/service/impl/PerformanceServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package solvd.ermakovich.ct.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import solvd.ermakovich.ct.domain.exception.EntityDoesNotExistException;
+import solvd.ermakovich.ct.domain.node.Performance;
+import solvd.ermakovich.ct.repository.PerformanceRepository;
+import solvd.ermakovich.ct.service.PerformanceService;
+import solvd.ermakovich.ct.web.dto.DancerInPerformance;
+
+/**
+ * @author Ermakovich Kseniya
+ */
+@Service
+@RequiredArgsConstructor
+public class PerformanceServiceImpl implements PerformanceService {
+
+    private final PerformanceRepository performanceRepository;
+
+    @Override
+    @Transactional(readOnly = true)
+    public Performance findById(String id) {
+        return performanceRepository.findById(id)
+                .orElseThrow(() -> new EntityDoesNotExistException("Performance does not exist"));
+    }
+
+    @Override
```

Comment:
@kermakovich did you miss `@Transactional` here?

Reply:
yes